src/mesh.cpp

```diff
@@ -2999,6 +3003,21 @@ void LibMesh::initialize()
   auto first_elem = *m_->elements_begin();
   first_element_id_ = first_elem->id();
 
+  // if the mesh is adaptive elements aren't guaranteed by libMesh to be
+  // contiguous in memory, so we need to map from bin indices (defined over
+  // active elements) to global dof ids
+  if (adaptive_) {
+    bin_to_elem_map_.reserve(m_->n_active_local_elem());
+    elem_to_bin_map_.resize(m_->n_local_elem(), 0);
```

Comment:
Is there any case where the default zeros could cause an issue? On the first iteration through the loop below, the first value that will be put into this map is indeed zero (`bin_to_elem_map_.size() -1` when the size is 1), which then seems confusing that we also have "empty" values that are zeros. Maybe making it -1 would more clearly indicate they are empty?